xrpl4j-model/src/test/java/org/xrpl/xrpl4j/model/immutables/FluentCompareToTest.java

```diff
@@ -0,0 +1,43 @@
+package org.xrpl.xrpl4j.model.immutables;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.primitives.UnsignedInteger;
+import org.junit.jupiter.api.Test;
+
+public class FluentCompareToTest {
+
+  static UnsignedInteger SMALLER = UnsignedInteger.valueOf(1);
+  static UnsignedInteger BIGGER = UnsignedInteger.valueOf(100);
+  static UnsignedInteger MAX_UINT = UnsignedInteger.MAX_VALUE;
+
+  @Test
+  public void simpleTest() {
+    assert(FluentCompareTo.is(SMALLER).equalTo(SMALLER));
+    assert(FluentCompareTo.is(BIGGER).equalTo(BIGGER));
+    assert(FluentCompareTo.is(SMALLER).notEqualTo(BIGGER));
+    assert(FluentCompareTo.isNot(SMALLER).notEqualTo(BIGGER));
+    assert(FluentCompareTo.isNot(SMALLER).equalTo(SMALLER));
+
+    assert(FluentCompareTo.is(SMALLER).lessThanOrEqualTo(SMALLER));
+    assert(FluentCompareTo.is(SMALLER).lessThanOrEqualTo(BIGGER));
+    assert(FluentCompareTo.is(SMALLER).lessThan(BIGGER));
+    assert(FluentCompareTo.is(SMALLER).notGreaterThan(BIGGER));
+    assert(FluentCompareTo.is(SMALLER).notGreaterThanEqualTo(BIGGER));
+
+    assert(FluentCompareTo.is(BIGGER).greaterThanEqualTo(BIGGER));
+    assert(FluentCompareTo.is(BIGGER).greaterThanEqualTo(SMALLER));
+    assert(FluentCompareTo.is(BIGGER).greaterThan(SMALLER));
```

Comment:
Can we add sameness checks for each value, e.g.:

```
assertThat(FluentCompareTo.is(SMALLER).isGreaterThan(SMALLER)).isNotTrue();
assertThat(FluentCompareTo.is(SMALLER).lessThan(SMALLER)).isNotTrue();
```

(unless I missed that somewhere else).